Added the bootstrap modal for creating a new room.

namrataaaa opened the pull request:
Changed the style.css file too and imported jquery and popper required for the data attributes of modal to function properly. Closes issue #118

namrataaaa committed:
Added the bootstrap modal for creating a new room. Changed the style.css file too and imported jquery and popper required for the data attributes of modal to function properly. Closes issue #118

LalitNM commented:
I am listing few changes required.

- You need to improve the UX of the modal(you can see #76, this pull request was also for adding modal)
A screenshot from #76.
![image](https://user-images.githubusercontent.com/63332774/95736673-0be28180-0c76-11eb-8561-61b6a73dd1ba.png)


- And when a room already exists your modal just collapses, it should show a message like shown in the screenshot below.

![image](https://user-images.githubusercontent.com/63332774/95736772-2f0d3100-0c76-11eb-950f-7c6346bfe4b5.png)

- In addition to this I think there are `.fadeIn` and `.fadeOut` in `client.js`. I don't think they are needed anymore.

LalitNM committed:
Update to .gitignore

Changes
- added .gitignore from templates provided by github
- added .DS_Store for macOS

LalitNM committed:
Fixed vulnerabilities

Changes
- update to package.json

LalitNM commented:
@namrataaaa I resolved conflicts, now you are good to go.

namrataaaa commented:
> @namrataaaa I resolved conflicts, now you are good to go.

Thank you. I have started working on the changes you requested.

namrataaaa committed:
Resolved conflict

namrataaaa committed:
Made the requested changes.

namrataaaa commented:
Hi, @LalitNM I have made the changes you requested. Improved the UX and the modal is showing the error message when a room already exists. However, after a room is being added, I am unable to dismiss the modal. I tried using ```modal('hide')``` function in client.js file but it did not work. Also, I have observed that the comparison of room names was case-sensitive. I tried to change it to uppercase and then compare but it also didn't work. I would also like to show a small notification to the user when a room is successfully created. Please help me on how I can achieve these changes.

LalitNM commented on app.js:
```diff
@@ -85,7 +85,7 @@ io.on('connection', function(socket) {
 
 			//if room name is taken
-			if(rooms[i].name == data.room_name) {
+			if(rooms[i].name.val().toUpperCase() === data.room_name.val().toUpperCase()) {
```
Any specific reason to change this? It is creating bug that modal doesn't collapses.

LalitNM commented on package.json:
```diff
@@ -20,8 +20,8 @@
   },
   "dependencies": {
+    "ejs": "^2.5.5",
     "express": "^4.14.0",
-    "socket.io": "^1.7.2",
-    "ejs": "^2.5.5"
+    "socket.io": "^2.3.0"
   }
-}
+}
```
Do not update these dependencies, because at present heroku doesn't support latest versions, app crashes when we update dependencies. You can remove this file.

LalitNM commented on public/script.js:
```diff
@@ -40,16 +40,13 @@ function search() {
-};
 
-$("#cancel").click(function() {
+$("#cancel").on('click',function() {
```
As far as I can see, you removed `#cancel` and `#sumbit`, you can remove this.

LalitNM commented on public/script.js:
```diff
@@ -40,16 +40,13 @@ function search() {
 
 function submitRoom() {
-  createRoom(); // used in client.js to pass details to app.js
+  createRoom(); // used in client.js to pass details to app.js 
```
Please try not to make some unnecessary changes.

LalitNM commented on public/style.css:
```diff
@@ -402,46 +377,17 @@ body {
+} */
 
-#roomName {
+/* #roomName {
   border: none;
-  border-bottom: 1.5px solid #000;
-}
+  border-bottom: 1px solid #000;
+} */
```
Can you remove this unnecessary commented code?

LalitNM commented on .gitignore:
```diff
@@ -1 +1,119 @@
+.next
+out
+
+# Nuxt.js build / generate output
+.nuxt
+dist
+
+# Gatsby files
+.cache/
+# Comment in the public line in if your project uses Gatsby and not Next.js
+# https://nextjs.org/blog/next-9-1#public-directory-support
+# public
+
+# vuepress build output
+.vuepress/dist
+
+# Serverless directories
+.serverless/
+
+# FuseBox cache
+.fusebox/
+
+# DynamoDB Local files
+.dynamodb/
+
+# TernJS port file
+.tern-port
+
+# Stores VSCode versions used for testing VSCode extensions
+.vscode-test
+
+# yarn v2
+.yarn/cache
+.yarn/unplugged
+.yarn/build-state.yml
+.yarn/install-state.gz
+.pnp.*
+
+# For macOS
+.DS_Store
```
This all is already there in master branch, you need to rebase your branch with our master branch(see instructions in contributing.md)

LalitNM commented on views/head_imports.ejs:
```diff
@@ -3,4 +3,8 @@
\ No newline at end of file
+<script src="/client.js"></script>
+
+<script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
+<script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js" integrity="sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI" crossorigin="anonymous"></script>
```
As you can see we are using cloudflare for cdns, please stick to this.

LalitNM commented on views/index.ejs:
```diff
@@ -1,96 +1,123 @@
-    <div class="wrapper">
-      <div class="app-container">
-        <div class="left">
-          <div class="form-group mb-0 px-3 pt-4">
-            <input type="text" id="searchId" placeholder="Search..." class="searchtext form-control" />
-            <hr class="mb-0"/>
-          </div>
-          <!-- Overflow style div -->
-          <div class="room-list">
-            <div class="card-columns m-2" name="people">
-              <div class="card active person" data-chat="lobby" id="lobby" onclick="showRoom(this)">
-                <div class="card-body">
-                  <h5 class="card-title name">Lobby</h5>
-                  <p class="card-text preview">Central Lobby</p>
-                </div>
+  <div class="wrapper">
+    <div class="app-container">
+      <div class="left">
+        <div class="form-group mb-0 px-3 pt-4">
+          <input type="text" id="searchId" placeholder="Search..." class="searchtext form-control" />
+          <hr class="mb-0" />
+        </div>
+        <!-- Overflow style div -->
+        <div class="room-list">
+          <div class="card-columns m-2" name="people">
+            <div class="card active person" data-chat="lobby" id="lobby" onclick="showRoom(this)">
+              <div class="card-body">
+                <h5 class="card-title name">Lobby</h5>
+                <p class="card-text preview">Central Lobby</p>
```
This all is also unnecessary changes, please use `save without formatting in your editor or change formatting rules of your editor.

LalitNM reviewed (changes requested):
I tried to answer all your questions in my review, only one regarding notification left, for this, do not add this functionality now, I will create a separate issue for this when somebody will fix #121.

namrataaaa committed:
Made the requested changes.

namrataaaa commented:
Hi, @LalitNM I made all the changes you requested and also rebased my branch on master. And I was finally able to dismiss the modal once a room was created. Thank you for suggesting those changes.

LalitNM commented:
@namrataaaa by mistake I deleted your files instead of editing them, can you run `git push -f origin feature-modal` on your system. This will restore the previous version of this pull request.

LalitNM committed:
removed trailing spaces

LalitNM committed:
removed unnecessary changes

LalitNM committed:
removed trailing space

LalitNM commented:
Merged :rocket: @namrataaaa. Good work on this.

namrataaaa commented:
> Merged 🚀 @namrataaaa. Good work on this.

Thanks, @LalitNM. I got to learn a lot and had a great experience. Can you please add the label ```hacktoberfest-accepted``` to this PR?

LalitNM commented:
I have already added `hacktoberfest` topic to whole repository. Still if you are saying, then here you go...
